llvm/cmake/modules/Findzstd.cmake

```diff
@@ -10,6 +10,35 @@
 # zstd::libzstd_shared
 # zstd::libzstd_static
 
+# Function to get zstd version.
+function(get_zstd_version_string zstd_INCLUDE_DIR OUT_VAR)
+
+    # Check if zstd.h file is present. Expectation is that this function will only
+    # be called if zstd is found and the include directory is valid.
+    if(NOT EXISTS "${zstd_INCLUDE_DIR}/zstd.h")
+      message(FATAL_ERROR "zstd.h not found in ${zstd_INCLUDE_DIR}. "
+        "Please set zstd_INCLUDE_DIR to the directory containing zstd.h.")
+    endif()
+
+    # Read the version string from zstd.h.
+    # The version is defined as macros ZSTD_VERSION_MAJOR, ZSTD_VERSION_MINOR,
+    # and ZSTD_VERSION_RELEASE in zstd.h.
+    file(READ "${zstd_INCLUDE_DIR}/zstd.h" content)
```

Comment:
does `zstd` not install a cmake file that we can use `find_package` with and get this info easily?

Reply:
got it, thanks